### PR TITLE
Fix `ignoreselected` typo in protocols.conf.erb template

### PR DESCRIPTION
**Pull request by VtG242**

### CODE REVIEW

**Review by kenyon — APPROVED** [review]

**Comment by igalic** [comment]
why are unit Tests failing on puppet 5 & 6?

File: templates/plugin/protocols.conf.erb [hunk 1]
```diff
@@ -9,8 +9,8 @@ if @values
   end
 end
 -%>
-<% if @ignore_selected != nil -%>
-  IgnoreSelected <%= @ignore_selected %>
+<% if @ignoreselected != nil -%>
```
Comment (alexjfisher) [inline templates/plugin/protocols.conf.erb:12]: Currently, `ignoreselected` is a Boolean.  That means it can never be `nil` (undef in puppet).

Comment (alexjfisher) [reply templates/plugin/protocols.conf.erb:12]: Looks like it was always a boolean and never default to `undef`.

**Review by bastelfreak — APPROVED** [review]

**Comment by alexjfisher** [comment]
@VtG242 Thanks for the bugfix!  I've made the existing test pass by changing the default to `undef` and added extra tests for the cases it's explicitly set to `true` or `false`.

**Pull request merged**